src/lib/safe-signature-verification/SafeSignatureVerifier.ts

```diff
@@ -0,0 +1,46 @@
+import { hashTypedData, type HashTypedDataParameters } from "viem";
+import { getRpcUrl } from "../../utils/getRpcUrl.js";
+import Safe from "@safe-global/protocol-kit";
+
+export default abstract class SafeSignatureVerifier {
+  protected chainId: number;
+  protected safeAddress: `0x${string}`;
+  protected rpcUrl: string;
+
+  constructor(chainId: number, safeAddress: `0x${string}`) {
+    const rpcUrl = getRpcUrl(chainId);
+
+    if (!rpcUrl) {
+      throw new Error(`Unsupported chain ID: ${chainId}`);
+    }
+    this.chainId = chainId;
+    this.safeAddress = safeAddress;
```

Comment:
verify if address is valid address using `getAddress` or `isAddress`. `getAddress` returns with the appropriate casing for consistency